packages/backend-ai/src/get_message.ts

```
@@ -0,0 +1,115 @@
+    modelId,
+    messages: messages.map((msg) => ({
+      role: msg.role,
+      content: msg.content.map(convertContentBlock),
+    })) as Message[],
+    system: systemPrompts,
+    ...(tools && tools.length > 0
+      ? {
+          toolConfig: {
+            tools: tools.map(convertTool),
+            toolChoice: { auto: {} },
+          },
+        }
+      : {}),
+  };
+
+  const command = new ConverseCommand(input);
+  const output = await client.send(command);
+
+  if (!output.output?.message) {
+    throw new Error('No message in ConverseCommandOutput');
+  }
+
+  return {
+    output: {
+      message: {
+        role: output.output.message.role as 'user' | 'assistant',
+        content: output.output.message.content as ContentBlock[],
+      },
+    },
+    stopReason: output.stopReason as GetMessageOutput['stopReason'],
+    usage: {
+      inputTokens: output.usage?.inputTokens ?? 0,
+      outputTokens: output.usage?.outputTokens ?? 0,
+      totalTokens: output.usage?.totalTokens ?? 0,
+    },
+    metrics: {
+      latencyMs: output.metrics?.latencyMs ?? 0,
+    },
+    additionalModelResponseFields: output.additionalModelResponseFields,
+  };
+};
+
+const convertContentBlock = (block: ContentBlock): ContentBlock => {
+  const result: Partial<ContentBlock> = {};
+
+  if ('text' in block && typeof block.text === 'string') {
+    result.text = block.text;
+  }
+  if ('image' in block && block.image) {
+    result.image = block.image;
+  }
+  if ('document' in block && block.document) {
+    result.document = block.document;
+  }
+  if ('toolUse' in block && block.toolUse) {
+    result.toolUse = block.toolUse;
+  }
+  if ('toolResult' in block && block.toolResult) {
+    result.toolResult = block.toolResult;
+  }
+  if ('guardContent' in block && block.guardContent) {
+    result.guardContent = block.guardContent;
+  }
+  if ('$unknown' in block && block.$unknown) {
+    result.$unknown = block.$unknown;
+  }
+
+  return result as ContentBlock;
```

Should it really just cast here or should we throw an error?